sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -144,23 +144,30 @@ Bigquery newBigqueryClient(String applicationName) {
         .build();
   }
 
+  @Nonnull
   @VisibleForTesting
   QueryResponse queryWithRetries(Bigquery bigqueryClient, QueryRequest queryContent,
                                  Sleeper sleeper, BackOff backOff)
       throws IOException, InterruptedException {
+    QueryResponse response;
     IOException lastException = null;
     do {
       try {
-        return bigqueryClient.jobs().query(projectId, queryContent).execute();
+        response = bigqueryClient.jobs().query(projectId, queryContent).execute();
+        if (response != null) {
+          return response;
+        }
```

Comment:
I think you can create and assign IOException to lastException with proper error message (based on what you know about null response).